docs/codeql/codeql-language-guides/extensible-predicates.rst

```diff
@@ -0,0 +1,182 @@
+.. _extensible-predicates:
+
+:orphan:
+:nosearch:
+
+Extensible predicates and their interaction with data extensions
```

Comment:
I need to spend more time looking at this article, but one thing that I'm wondering is whether the general information about extensible predicates should live in the in the language reference. Reasoning: extensible predicates are now a general, language-agnostic thing in CodeQL which in theory could be created by any author to allow structured data to be added to an analysis. For CodeQL model packs, we just happen to have created three 'well-known' ones for sources, sinks, and summaries that are used for modeling dataflow through frameworks in Java. Creating a new reference article might be beyond the scope of this release though.

Reply:
That makes sense. As I was looking at this I thought that whatever we could make generic would be helpful when we add the next language.

Reply:
Comment added to follow up issue.